src/gcp/location.ts

```diff
@@ -0,0 +1,112 @@
+// A mapping from geographical region to subdomain, usefull for Container Registry
+const SUBDOMAIN_MAPPING: Record<string, string> = {
+  "us-west1": "us",
+  "us-west2": "us",
+  "us-west3": "us",
+  "us-west4": "us",
+  "us-central1": "us",
+  "us-central2": "us",
+  "us-east1": "us",
+  "us-east4": "us",
+  "northamerica-northeast1": "us",
+  "southamerica-east1": "us",
+  "europe-west1": "eu",
+  "europe-west2": "eu",
+  "europe-west3": "eu",
+  "europe-west4": "eu",
+  "europe-west5": "eu",
+  "europe-west6": "eu",
+  "europe-central2": "eu",
+  "europe-north1": "eu",
+  "asia-east1": "asia",
+  "asia-east2": "asia",
+  "asia-northeast1": "asia",
+  "asia-northeast2": "asia",
+  "asia-northeast3": "asia",
+  "asia-south1": "asia",
+  "asia-southeast2": "asia",
+  "australia-southeast1": "asia",
+};
+
+/** Google Cloud Storage multi-region mapping (https://cloud.google.com/storage/docs/locations#location-mr) */
+const MULTI_REGION_MAPPING: Record<string, string> = {
+  /** us */
+  "us-central1": "us",
+  "us-east1": "us",
+  "us-east4": "us",
+  "us-west1": "us",
+  "us-west2": "us",
+  "us-west3": "us",
+  "us-west4": "us",
+  /** eu */
+  "europe-central2": "eu",
+  "europe-north1": "eu",
+  "europe-west1": "eu",
+  "europe-west3": "eu",
+  "europe-west4": "eu",
+  "europe-west5": "eu",
+  /** asia */
+  "asia-east1": "asia",
+  "asia-east2": "asia",
+  "asia-northeast1": "asia",
+  "asia-northeast2": "asia",
+  "asia-northeast3": "asia",
+  "asia-south1": "asia",
+  "asia-south2": "asia",
+  "asia-southeast1": "asia",
+  "asia-southeast2": "asia",
+};
+
+/** Google Cloud Storage dual-region mapping (https://cloud.google.com/storage/docs/locations#location-dr) */
+const DUAL_REGION_MAPPING: Record<string, string> = {
+  /** asia1 */
+  "asia-northeast1": "asia1",
+  "asia-northeast2": "asia1",
+  /** eur4 */
+  "europe-north1": "eur4",
+  "europe-west4": "eur4",
+  /** nam4 */
+  "us-central1": "nam4",
+  "us-east1": "nam4",
+};
+
+/**
+ * Obtains the subdomains for all the hostname options in Container Registry
+ */
+export function getContainerRegistrySubdomains(): string[] {
```

Comment:
What is the point of these trivial functions? So we can stub them in tests?

Reply:
In the discussion above with @taeold, we thought it would be best to expose a helper function in case we ever had to change the underlying data structure. Would you prefer that we just expose the actual constants?

Reply:
I think for a public API that level of defensiveness can make sense. For an internal codebase it may be overly defensive.

Reply:
Removed all getters & exported the constants directly